### ISSUE TITLE
Support literal dictionaries with symbolic keys

### ISSUE THREAD

**Issue by pschanely** [pschanely/CrossHair#154]
Similar to #114, literal map constructions with symbolic keys will not be analyzable.

We'll need to intercept the BUILD_MAP and BUILD_CONST_KEY_MAP opcodes and upgrade the return to a symbolic dictionary.

**Comment by pschanely** [comment]
Note: here is an example where we'd like to be able to find the counterexample: https://crosshair-web.org/?crosshair=0.1&python=3.8&gist=79cb9a330f194e444a7c64c727fed781

This **might** be an appropriate hacktoberfest issue for someone that already knows about CPython bytecode, or wants to learn about this.

**Comment by aksh16** [comment]
Commenting here so Github can let me become the assignee

**Comment by aksh16** [comment]
Thank you so much for the example! Also, your explanation cleared up a lot of things. I might have more questions once I start with the implementation but I got the gist of what needs to be done.

**Comment by pschanely** [comment]
> Thank you so much for the example! Also, your explanation cleared up a lot of things. I might have more questions once I start with the implementation but I got the gist of what needs to be done.

Yes, please be liberal with questions - here or discord is fine!
And thanks again - this enhancement is a relatively large step forward in CrossHair's ability to retain symbolics.
